src/implementation/Client/HTTPClient/pubsub.ts

```diff
@@ -25,9 +27,11 @@ export default class HTTPClientPubSub implements IClientPubSub {
     this.logger = new Logger("HTTPClient", "PubSub", client.getOptions().logger);
   }
 
-  async publish(pubSubName: string, topic: string, data: object = {}): Promise<boolean> {
+  async publish(pubSubName: string, topic: string, data: object = {}, metadata?: KeyValueType): Promise<boolean> {
+    const queryParams = getHTTPMetadataQueryParam(metadata ?? {});
```

Comment:
Shouldn't we just pass metadata and make it default in the `getHTTPMetadataQueryParam`?

Reply:
But it defaults through the method signature that adds a ? No?

Reply:
I see where you are coming from, sure! Let me update it.